File: pet-adoption/frontend/pet-pal-finder/src/app/service/rating.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { Rating, StatusMessage } from '../models/models';

@Injectable({
  providedIn: 'root'
})
export class RatingService {

 
  private baseUrl: string = environment.baseUrl;

  constructor(private http: HttpClient) { }

  addRating(rating:Rating){
    return this.http.post<StatusMessage>(
      `${this.baseUrl}rating`,rating
    );
  }
}

File: pet-adoption/backend/petPalFinder/src/main/java/com/petAdoption/petPalFinder/services/RatingService.java
package com.petAdoption.petPalFinder.services;

import java.util.List;

import com.petAdoption.petPalFinder.dto.StatusMessage;
import com.petAdoption.petPalFinder.models.Rating;

public interface RatingService {
    public List<Rating> getDoctor();
    public StatusMessage addRating(Rating rating);
}
